src/ResourceManager/CognitiveServices/Commands.Management.CognitiveServices/CognitiveServicesAccount/NewAzureCognitiveServicesAccount.cs

```diff
@@ -0,0 +1,135 @@
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "Cognitive Services Account Name.")]
+        [Alias(CognitiveServicesAccountNameAlias, AccountNameAlias)]
+        [ValidateNotNullOrEmpty]
+        [ValidatePattern("^[a-zA-Z0-9][a-zA-Z0-9_.-]*$")]
+        [ValidateLength(3, 24)]
+        public string Name { get; set; }
+
+        [Parameter(
+            Position = 2,
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "Cognitive Services Account Type.")]
+        [Alias(CognitiveServicesAccountTypeAlias, AccountTypeAlias, KindAlias)]
+        [ValidateSet(
+            AccountType.ComputerVision,
+            AccountType.Emotion,
+            AccountType.Face,
+            AccountType.LUIS,
+            AccountType.Recommendations,
+            AccountType.Speech,
+            AccountType.TextAnalytics,
+            AccountType.WebLM,
+            IgnoreCase = true)]
+        public string Type { get; set; }
+
+        [Parameter(
+            Position = 3,
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "Cognitive Services Account Sku Name.")]
+        [ValidateSet(
+            AccountSkuString.F0, 
+            AccountSkuString.S0, 
+            AccountSkuString.S1, 
+            AccountSkuString.S2, 
+            AccountSkuString.S3, 
+            AccountSkuString.S4, 
+            IgnoreCase = true)]
+        public string SkuName { get; set; }
+
+        [Parameter(
+            Position = 4,
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "Cognitive Services Account Location.")]
+        [ValidateNotNullOrEmpty]
+        public string Location { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "Cognitive Services Account Tags.")]
+        [Alias (TagsAlias)]
+        [ValidateNotNull]
+        [AllowEmptyCollection]
+        public Hashtable[] Tag { get; set; }
+
+        [Parameter(Mandatory = false, HelpMessage = "Don't ask for confirmation.")]
+        public SwitchParameter Force { get; set; }
+
+        public override void ExecuteCmdlet()
+        {
+            base.ExecuteCmdlet();
+
+            CognitiveServicesAccountCreateParameters createParameters = new CognitiveServicesAccountCreateParameters()
+            {
+                Location = this.Location,
+                Kind = ParseAccountKind(this.Type).Value, // must have value, mandatory parameter
+                Sku = new Sku(ParseSkuName(this.SkuName)),
+                Tags = TagsConversionHelper.CreateTagDictionary(Tag, validate: true),
+                Properties = new object(), // Must not be null according to Azure RM spec. Also there is no actual properties to pass, so not exposing it through cmdlet
+            };
+
+            this.ConfirmAction(
```

Comment:
Two things:
1. Unfortunately, this common piece of code does not currently implement ShouldProcess correctly.  
2. Generally, you do not need a Force parameter for New cmdlets, ```Instead, simple use if (ShouldProcess(target, action) {... actions...}```